File: Python/project07.py
#ProjectEuler Number 7
#by Josh Smith

import math
def is_prime(n):
    if n % 2 == 0 and n > 2: 
        return False
    return all(n % i for i in range(3, int(math.sqrt(n)) + 1, 2))

bool = False
i = 2
j = 0
while j < 10001:
	if is_prime(i):
		j += 1
	i += 1
	
print i-1

File: C++/project07.cpp
//ProjectEuler Number 7
//by Josh Smith

#include <iostream>
using namespace std;

bool isPrime(size_t n){
	if(n < 0){
		cout << "n must be larger than zero" << endl;
		return false;
	}

	for(int i = 2; i < sqrt(n)+1; i++){
		if(n%i == 0){
			return false;
		}
	}
	return true;
}

int main() {
	size_t i = 0;
	size_t x = 2;
	while(i <= 10001){
		if(isPrime(x)){
			i++;
		}
		x++;
	}
	cout << x-1 << endl;
	return 0;
}
